### PR TITLE
Support PEtabs initializationPrior fix #534

### LINKED ISSUE #534: Support PEtabs initializationPrior

### PR DESCRIPTION
**Pull request by jvanhoefer**

### CODE REVIEW

**Comment by jvanhoefer** [comment]
Ok, I will have a look at the Code Coverage after lunch... :)

**Comment by jvanhoefer** [comment]
> Ok, I will have a look at the Code Coverage after lunch... :)

Seems to be a bug in the first image shown. So everything seems fine from that side :) (Tested it locally on an example)

File: pypesto/petab/importer.py [hunk 1]
```diff
@@ -304,6 +304,25 @@ def create_prior(self) -> NegLogParameterPriors:
...
+    def create_startpoint_method(self):
+        """
+        Creates a startpoint method, if the PEtab problem specifies an
+        initializationPrior. Returns None, if no initializationPrior
+        is specified.
+        """
+        if petab.INITIALIZATION_PRIOR_TYPE \
+                not in self.petab_problem.parameter_df:
+            return None
+        else:
```
Comment (dweindl) [inline pypesto/petab/importer.py:316]: Remove unnecessary `else`after `return`

File: pypesto/problem.py [hunk 2]
```diff
@@ -49,6 +50,8 @@ class Problem:
     x_guesses:
         Guesses for the parameter values, shape (g, dim), where g denotes the
         number of guesses. These are used as start points in the optimization.
+    startpoint_method:
+        Method used for generating initial values for the optimization.
```
Comment (dweindl) [inline pypesto/problem.py:54]: Please document expected arguments / return type

File: pypesto/petab/importer.py [hunk 3]
```diff
@@ -304,6 +304,25 @@ def create_prior(self) -> NegLogParameterPriors:
...
+            def startpoint_method(n_samples: int, **kwargs):
```
Comment (dweindl) [inline pypesto/petab/importer.py:317]: `kwargs` intentionally unused?

**Review by dweindl — APPROVED** [review]
Looks good as far as I can tell. :+1:

File: pypesto/petab/importer.py (hunk 3 above)
Comment (jvanhoefer) [reply pypesto/petab/importer.py:317]: yep. This was included, to make the interface consistent with the other start point methods, which have different key word arguments, depending on the method.

Comment (jvanhoefer) [reply pypesto/petab/importer.py]: (Which again is needed inside of `assign_startpoints`, that uses an unified interface for all those methods... as far as I understand)

**Review by FFroehlich — APPROVED** [review]

File: pypesto/optimize/optimize.py [hunk 4]
```diff
@@ -64,7 +64,14 @@ def minimize(
         optimizer = ScipyOptimizer()
 
     # startpoint method
-    if startpoint_method is None:
+    if (startpoint_method is not None) \
```
[the remark this answers is not in the document]
Comment (jvanhoefer) [reply pypesto/optimize/optimize.py:67]: :+1: I will change the `Error` out a warning.

File: pypesto/petab/importer.py [hunk 5]
```diff
@@ -304,6 +304,25 @@ def create_prior(self) -> NegLogParameterPriors:
...
+        """
+        Creates a startpoint method, if the PEtab problem specifies an
+        initializationPrior. Returns None, if no initializationPrior
+        is specified.
+        """
+        if petab.INITIALIZATION_PRIOR_TYPE \
+                not in self.petab_problem.parameter_df:
+            return None
+
+        def startpoint_method(n_samples: int, **kwargs):
```
Comment (dweindl) [inline pypesto/petab/importer.py:317]: @jvanhoefer : shouldn't that be `n_starts`? Seems like that is what's expected in `pesto.startpoint.util.assign_startpoints`.

Comment (jvanhoefer) [reply pypesto/petab/importer.py:317]: yep

**Pull request merged**